packages/core/src/local-admin-api.ts

```diff
@@ -20,30 +20,26 @@ export class LocalAdminApi implements AdminApi {
     return this.nodeStatusFn()
   }
 
-  async startIndexingModels(
-    modelsIDs: Array<StreamID>,
-    indices?: Array<ModelFieldsIndex>
-  ): Promise<void> {
-    let models = modelsIDs.map((id) => {
-      return {
-        streamID: id,
-        indices: undefined,
-      } as ModelFieldsIndex
-    })
-    if (indices) {
-      models = models.concat(indices)
+  async startIndexingModels(modelsIDs: Array<StreamID>, indices?: Array<ModelData>): Promise<void> {
+    const models = indices ?? []
+    for (const modelId of modelsIDs) {
```

Comment:
this still feels really ugly to me, but if we split up this method into two versions like I suggested elsewhere then the need for this goes away